store/src/java/com/zimbra/cs/event/logger/EventLogger.java

```diff
@@ -197,29 +205,29 @@ private EventNotifier(Map<String, EventLogHandler.Factory> knownFactories, Map<S
         @Override
         public void run() {
             try {
-                while (!Thread.currentThread().isInterrupted()) {
-                    consume(eventQueue);
+                Event event = null;
+                while (!shutdownExecutor.get()) {
+                    event = eventQueue.poll(3, TimeUnit.SECONDS);
+                    if (event != null && event.getEventType() != Event.EventType.POISON_PILL) {
```

Comment:
Nitpick: import `EventType` directly so you don't need to keep repeating `Event.EventType.POISON_PILL`